test/extended/util/oauthserver/oauthserver.go

```diff
@@ -0,0 +1,546 @@
+
+	oauthConfigPath  = "/var/config/system/configmaps/oauth-config"
+	serviceCADirPath = "/var/config/system/configmaps/service-ca"
+
+	configObjectsDir = "/var/oauth/configobjects/"
+
+	RouteName = "test-oauth-route"
+	SAName    = "e2e-oauth"
+)
+
+var (
+	serviceCAPath = "/var/config/system/configmaps/service-ca/service-ca.crt" // has to be var so that we can use its address
+
+	defaultProcMount         = corev1.DefaultProcMount
+	volumesDefaultMode int32 = 420
+)
+
+// DeployOAuthServer - deployes an instance of an OpenShift OAuth server
+// very simplified for now
+// returns OAuth server url, cleanup function, error
+func DeployOAuthServer(oc *exutil.CLI, idps []osinv1.IdentityProvider, configMaps []corev1.ConfigMap, secrets []corev1.Secret) (*tokencmd.RequestTokenOptions, func(), error) {
+	oauthServerDataDir := exutil.FixturePath("testdata", "oauthserver")
+	cleanups := func() {
+		oc.AsAdmin().Run("delete").Args("clusterrolebinding", oc.Namespace()).Execute()
+		oc.AsAdmin().Run("delete").Args("oauthclient", oc.Namespace()).Execute()
+	}
+
+	if err := oc.AsAdmin().Run("create").Args("-f", path.Join(oauthServerDataDir, "oauth-sa.yaml")).Execute(); err != nil {
+		return nil, cleanups, err
+	}
+
+	// the oauth server needs access to kube-system configmaps/extension-apiserver-authentication
+	oauthSARolebinding := &rbacv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: oc.Namespace(),
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: rbacv1.GroupName,
+			Kind:     "ClusterRole",
+			Name:     "cluster-admin",
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      SAName,
+				Namespace: oc.Namespace(),
+			},
+		},
+	}
+	if _, err := oc.AdminKubeClient().RbacV1().ClusterRoleBindings().Create(oauthSARolebinding); err != nil {
+		return nil, cleanups, err
+	}
+
+	for _, res := range []string{"cabundle-cm.yaml", "oauth-server.yaml"} {
+		if err := oc.AsAdmin().Run("create").Args("-f", path.Join(oauthServerDataDir, res)).Execute(); err != nil {
+			return nil, cleanups, err
+		}
+	}
+
+	// create the secrets and configmaps the OAuth server config requires to get the server going
+	coreClient := oc.AdminKubeClient().CoreV1()
+	cmClient := coreClient.ConfigMaps(oc.Namespace())
+	secretsClient := coreClient.Secrets(oc.Namespace())
+
+	for _, cm := range configMaps {
+		if _, err := cmClient.Create(&cm); err != nil {
+			return nil, cleanups, err
+		}
+	}
+
+	for _, secret := range secrets {
+		if _, err := secretsClient.Create(&secret); err != nil {
+			return nil, cleanups, err
+		}
+	}
+
+	// generate a session secret for the oauth server
+	sessionSecret, err := randomSessionSecret()
+	if err != nil {
+		return nil, cleanups, err
+	}
+	if _, err := secretsClient.Create(sessionSecret); err != nil {
+		return nil, cleanups, err
+	}
+
+	// get the route of the future OAuth server
+	route, err := oc.AdminRouteClient().RouteV1().Routes(oc.Namespace()).Get(RouteName, metav1.GetOptions{})
```

Comment:
Where did this get created?

Reply:
it's in test/extended/testdata/oauth-server.yaml

Reply:
The file is now `test/extended/testdata/oauthserver/oauth-network.yaml` and I added a comment about what it's creating.